fix #610 privacy CA now known as attestation CA

equalsJeffH opened the pull request:
fixes #610

[ in case anyone noticed (unlikely, I realize): the branch name this PR is based on is incorrect, it should be `jeffh-fix-610-privacy-ca-cites`, but oh well it is not. too much trouble to change it at this point...]


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/762.html" title="Last updated on Jan 30, 2018, 11:45 PM GMT (b380a03)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/762/b0cae5e...b380a03.html" title="Last updated on Jan 30, 2018, 11:45 PM GMT (b380a03)">Diff</a>

emlun commented on index.bs:
```diff
@@ -2645,12 +2646,14 @@ WebAuthn supports multiple attestation types:
+:: In this case, an [=authenticator=] is based on a Trusted Platform Module (TPM) and holds an authenticator-specific
+    "endorsement key" (EK). This key is used to securely communicate with a trusted third party, the [=Attestation CA=]
+    [[!TCG-CMCProfile-AIKCertEnroll]] (formerly known as a "Privacy CA"). The [=authenticator=] can generate multiple
+    attestation identity key pairs (AIK) and requests an [=Attestation CA=] to issue an attestation identity certificate (AIK)
```
Should the latter AIK here be AIC?

equalsJeffH replied:
"AIC" is not used in [[!TCG-CMCProfile-AIKCertEnroll]], so I'm thinking not....

emlun replied:
Ok, I just wanted to thought it looked strange that the same acronym was used for two different things.

equalsJeffH replied:
thx, you did identify a problem -- should be using term "AIK certificate" here.

emlun commented on index.bs:
```diff
@@ -2645,12 +2646,14 @@ WebAuthn supports multiple attestation types:
+    [[!TCG-CMCProfile-AIKCertEnroll]] (formerly known as a "Privacy CA"). The [=authenticator=] can generate multiple
+    attestation identity key pairs (AIK) and requests an [=Attestation CA=] to issue an attestation identity certificate (AIK)
+    for each. Using this approach, such an [=authenticator=] can limit the exposure of the EK (which is a global correlation
+    handle) to Attestation CA(s). AIKs can be requested for each [=authenticator=]-generated [=public key credential=]
```
Ditto

equalsJeffH replied:
Ditto ;-)

equalsJeffH replied:
I am thinking usage of AIK here is nominally correct.

emlun replied:
Yeah, this looks better!

selfissued commented on index.bs:
```diff
@@ -1746,14 +1746,15 @@ during credential generation.
         The client MAY replace the [=AAGUID=] and [=attestation statement=] generated
         by the authenticator with meaningless client-generated values. For example, in order to avoid having to obtain
-        [=user consent=] to relay uniquely identifying information to the [=[RP]=], or to save a roundtrip to a Privacy CA.
+        [=user consent=] to relay uniquely identifying information to the [=[RP]=], or to save a roundtrip to a
```
Change "a" to "an"

equalsJeffH replied:
fixed, see https://github.com/w3c/webauthn/pull/762/commits/0cbccd00839e874b81409414256c6071e322f6aa

jcjones commented:
It looks like the CI build didn't run... is Anonymization CA defined anywhere? I don't see it.

equalsJeffH commented:
@jcjones 
> is Anonymization CA defined anywhere? I don't see it.

see https://github.com/w3c/webauthn/pull/762/files#diff-ec9cfa5f3f35ec1f84feb2e59686c34dR4448

> It looks like the CI build didn't run...

it appears it was that "used only once" `allowCredentialDescriptorList` warning that we fixed earlier today. now fixed in master and this branch.  looks like travisCI is now running....

thx for headzup :)

equalsJeffH commented:
bummer.  

> FATAL ERROR: Couldn't find 'TCG-CMCProfile-AIKCertEnroll' in bibliography data. 

I don't know what's going.....wait...

https://www.specref.org/?q=TCG-CMCProfile-AIKCertEnroll

....shows that the reference is in specref.org...so i think this error is a hint that we need to update our cached repo-side build stuff such that we have the latest specref data ?

jcjones commented:
@equalsJeffH: I've updated the spec-data, but to tell in the PR we'd need to rebase the PR, which I can't do from this interface.

equalsJeffH commented:
@jcjones: I've updated the spec-data...
thanks!  I merged-from-master https://github.com/w3c/webauthn/pull/762/commits/cb6ba298190894cbfc01ba6779921f2d5860d36b ...waiting to see if the travis build is successful...

equalsJeffH commented:
it would be nice to get an OK from Chrome folk since I invented the "anonymization CA" name for the entity they have so far been refering to as a "Privacy CA" (which would end up being confusing since that name was originally used by TCG specs for an entity that behaves somewhat differently and now is itself renamed to "Attestation CA")

agl commented:
If we end up using a privacy CA, we're probably going to call it a privacy CA irrespective of whatever is in the spec. I think the term is pretty well established for "thing that checks a specific certificate and issues a general one saying that it checked it". I don't see that we need another term for the same thing.

equalsJeffH commented:
@agl wrote:
> If we end up using a privacy CA, we're probably going to call it a privacy CA irrespective of whatever is in the spec.

Thx, understood. I prefer to keep these terms separate in the spec, and y'all can call such functionality whatever you want, if it is materialized.

selfissued commented:
Agreed.  I think this is ready to merge.
